### PR TITLE
Refactor/migrate other stuff to stores

### PR DESCRIPTION
**Pull request by abegong**
This PR should be merged after #691.

This PR should NOT be merged until the FIXMEs are resolved, especially the ones on SiteIndexPageRenderer.

### CODE REVIEW

File: great_expectations/render/renderer/new_site_builder.py [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+        self.view_class = getattr(view_module, view.pop("class_name"))
+
+
+    def build(self):
+        # Loop over sections in the HtmlStore
+        logger.debug("DefaultSiteIndexBuilder.build")
+
+        resource_keys = self.target_store.list_keys()
+
+        # FIXME : There were no tests to verify that content is created or correct,
```
Comment (abegong) [inline great_expectations/render/renderer/new_site_builder.py:245]: @roblim These five lines are the place where I need some help. To activate these, we'll need to refactor `SiteIndexPageRenderer`

**Comment by abegong** [comment]
@jcampbell Ready for preliminary review

File: great_expectations/data_context/data_context.py [hunk 2]
```diff
@@ -24,7 +24,10 @@
 
 from great_expectations.exceptions import DataContextError, ConfigNotFoundError, ProfilerError
 
-from great_expectations.render.renderer.site_builder import SiteBuilder
+# FIXME : fully deprecate site_builder, by replacing it with new_site_builder.
```
Comment (jcampbell) [inline great_expectations/data_context/data_context.py:27]: I'm holding you to all the FIXME and TODO items...

Comment (jcampbell) [reply great_expectations/data_context/data_context.py:27]: :)

File: great_expectations/data_context/store/evaluation_parameter_store.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+class EvaluationParameterStore(object):
```
Comment (jcampbell) [inline great_expectations/data_context/store/evaluation_parameter_store.py:1]: Doesn't this need run_id in its key?

**Review by jcampbell — APPROVED** [review]
I'm approving this but noting that it essentially makes us track a lot more in the single branch in terms of todos. Excited for more refactors!

**Pull request merged**